Mismatch types on arm device

```
error[E0308]: mismatched types
   --> /home/op/.cargo/registry/src/github.com-1ecc6299db9ec823/socketcan-alt-0.2.2/src/socket.rs:124:50
    |
124 |             (*msg.as_mut_ptr()).msg_controllen = cmsg_buf.len();
    |                                                  ^^^^^^^^^^^^^^ expected `u32`, found `usize`
    |
help: you can convert an `usize` to `u32` and panic if the converted value wouldn't fit
    |
124 |             (*msg.as_mut_ptr()).msg_controllen = cmsg_buf.len().try_into().unwrap();
    |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

This happened on `armv7-unknown-linux-musleabihf`

--- a/.github/workflows/rust.yaml
+++ b/.github/workflows/rust.yaml
@@ -17,7 +17,7 @@ jobs:
     - uses: actions/checkout@v2
     - run: rustup target add ${{ matrix.target }}
     - run: cargo build --verbose --target ${{ matrix.target }} --all-targets --all-features
-    - run: cargo test --verbose --target ${{ matrix.target }}
+    - run: cargo test --verbose --target ${{ matrix.target }} --all-features
   lint:
     runs-on: ubuntu-latest
     steps:

--- a/.github/workflows/rust.yaml
+++ b/.github/workflows/rust.yaml
@@ -8,10 +8,16 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+        - x86_64-unknown-linux-gnu
+        - x86_64-unknown-linux-musl
     steps:
     - uses: actions/checkout@v2
-    - run: cargo build --verbose --all-targets --all-features
-    - run: cargo test --verbose
+    - run: rustup target add ${{ matrix.target }}
+    - run: cargo build --verbose --target ${{ matrix.target }} --all-targets --all-features
+    - run: cargo test --verbose --target ${{ matrix.target }}
   lint:
     runs-on: ubuntu-latest
     steps:

--- a/src/socket.rs
+++ b/src/socket.rs
@@ -121,7 +121,7 @@ impl Socket {
             (*msg.as_mut_ptr()).msg_iov = iov.as_mut_ptr();
             (*msg.as_mut_ptr()).msg_iovlen = 1;
             (*msg.as_mut_ptr()).msg_control = cmsg_buf.as_mut_ptr() as _;
-            (*msg.as_mut_ptr()).msg_controllen = cmsg_buf.len();
+            (*msg.as_mut_ptr()).msg_controllen = cmsg_buf.len() as _;
 
             let size = libc::recvmsg(self.as_raw_fd(), msg.as_mut_ptr(), 0);
             // frame will be moved

